### PR TITLE
[NONMODULAR]Demonic Watcher Nerfs&Fixes

### PR DESCRIPTION
**Pull request by Vizoplez**
## About The Pull Request
Demonic watchers will no longer teleport from 3 tiles away
Demonic watchers will no longer aggro from off-screen or through walls
Demonic watchers no longer passively regenerate HP as long as they do not have a 'target', which is actually inapplicable due to that condition only being met when they are within 3 tiles of a person
Demonic watchers now no longer drop your temperature by -70
Demonic watchers now also do more damage in exchange for aformentioned nerfs
Demonic watchers are no longer semi-megafauna due to their AI freaking out, causing them to teleport too often; becoming unkillable bar the AI lagging behind.

## Why It's Good For The Game
Why god why /tg/.

## Changelog
:cl:
tweak: Massive nerfs to demonic ice-watchers to balance them in line with other fauna. Come on.
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
bad PR , they are only on the last Z-level where most of the ores are , if you don't want to deal with them then don't go there , however they are not unkillable , and certainly not fauna if you got a hardsuit with heating.
Also make them not drop bluespace cores if this nef even goes through.. it'd be far too easy to kill them now.
Also keep in mind we have 135 health , compared to the TG 100 health , you don't even need miner gear to defeat them as long as you got a ranged weapon.

**Comment by Vizoplez** [comment]
> bad PR , they are only on the last Z-level where most of the ores are , if you don't want to deal with them then don't go there , however they are not unkillable , and certainly not fauna if you got a hardsuit with heating.
> Also make them not drop bluespace cores if this nef even goes through.. it'd be far too easy to kill them now.
> Also keep in mind we have 135 health , compared to the TG 100 health , you don't even need miner gear to defeat them as long as you got a ranged weapon.

This is quite false for a variety of reasons. One; miners do NOT use hardsuits with heating down there for the most part, the most common armors are as follows;
Ash Drake Armor
Explorer Suits
NarSian robes

This is due to level 3 having full pressure; and therefor slowing hardsuits down in it.
Furthermore; 'its supposed to be hard' is not an excuse for having an AI that can off-screen see the player, through walls; Thats not good gameplay. Even with miner gear they were near impossible to kill due to how their AI decides to jump if you get within 3 tiles of them; or if you get too far, every 1.5 seconds. Demonic watchers were poorly designed; and can still quite easily kill you. Its a substantial nerf but still makes them an absolute menace if mishandled.

**Comment by SpaceVampire** [comment]
This really seems like a salt PR.
I have spent a decent amount of time as a shaft miner on Ice Box.
I had some trouble fighting them with the crusher at first.
But never any trouble just blasting em with the PKA.
The third level is supposed to be pretty scary, and they only spawn on it.
So if you are fighting them in the first place, you understand the risk.
I feel like you went way too overboard on the nerfs.
Maybe try PRing this upstream instead, to see how they feel about it?

EDIT: Looking at your exact values, you turned them into a total joke of a mob that isn't going to kill a single person.
It was worse than I thought at first glance.

**Comment by Draganfrukts** [comment]
Their object damage was fine as-is tbh

**Comment by NullDagaf** [comment]
From a crusher-main experience.
Watchers were really only an issue if there were more than two of them, say near a demonic portal.
Otherwise, there was a pattern to follow. Wait for them to teleport, dodge the icebolt, go in for attack as they move closer to you.  Repeat twice, and they're dead. You don't even need to mark the second attack, only one is nessesary. They don't move unpredictably or anything, heck, they WILL line up with you cardinally if you give them room for that.

Now, commenting on the changes.
First off, they only shoot after teleporting. There's absolutely no point to change ranged_cooldown_time. (Also, will probably cause issues for miners, having been used to the normal watchers)
Second. You DO realise that by changing the ice bolt, you're also changing the regular watchers' attack, yes? The lil' winged shit that's already useless on it's own and whose bolts are the only way of attacking proper.
Third. Making the delay for movement bigger is already a massive nerf. As it was, the watcher could barely outrun you, which meant you can't just go "haha crusher go brrr" and run at them, same with unupgraded KA's. You actually had to wait until it got closer to you, THEN you could try attacking it. By making the delay bigger? You just completely threw that out of the window.
Fourth. Teleport being as close as is, will mean it will basically not teleport from a KA user. Yes, the retreat_distance got lowered to 1. But the move_delay got raised too, so you could literally just run away and it wouldn't do a thing, as again. It doesn't shoot unless it teleports.


TL:DR. "PR bad, just git gud @Yawet330 "

**Comment by NullDagaf** [comment]
> bad PR , they are only on the last Z-level where most of the ores are , if you don't want to deal with them then don't go there , however they are not unkillable , and certainly not fauna if you got a hardsuit with heating.
> Also make them not drop bluespace cores if this nef even goes through.. it'd be far too easy to kill them now.
> Also keep in mind we have 135 health , compared to the TG 100 health , you don't even need miner gear to defeat them as long as you got a ranged weapon.

Also, what Yawet said. No sane miner uses a hardsuit.
...maybe if you get one from the tendrils, but even then ash drake armor is better.

~~**prateor suit doesn't count**~~

**Comment by dawsonkeyes** [comment]
> From a crusher-main experience.
> Watchers were really only an issue if there were more than two of them, say near a demonic portal.
> Otherwise, there was a pattern to follow. Wait for them to teleport, dodge the icebolt, go in for attack as they move closer to you. Repeat twice, and they're dead. You don't even need to mark the second attack, only one is nessesary. They don't move unpredictably or anything, heck, they WILL line up with you cardinally if you give them room for that.
> 
> Now, commenting on the changes.
> First off, they only shoot after teleporting. There's absolutely no point to change ranged_cooldown_time. (Also, will probably cause issues for miners, having been used to the normal watchers)
> Second. You DO realise that by changing the ice bolt, you're also changing the regular watchers' attack, yes? The lil' winged shit that's already useless on it's own and whose bolts are the only way of attacking proper.
> Third. Making the delay for movement bigger is already a massive nerf. As it was, the watcher could barely outrun you, which meant you can't just go "haha crusher go brrr" and run at them, same with unupgraded KA's. You actually had to wait until it got closer to you, THEN you could try attacking it. By making the delay bigger? You just completely threw that out of the window.
> Fourth. Teleport being as close as is, will mean it will basically not teleport from a KA user. Yes, the retreat_distance got lowered to 1. But the move_delay got raised too, so you could literally just run away and it wouldn't do a thing, as again. It doesn't shoot unless it teleports.
> 
> TL:DR. "PR bad, just git gud @Yawet330 "

Basically everything that was said here. This seems like a PR from someone that would rather not deal with the watchers altogether instead of actually trying to click on them- they’re not difficult enemies by any means, and nerfing them in such a manner feels like a salt PR to me.

**Comment by Gandalf2k15** [comment]
Please, if there is a balance issue, take it upstream, see how they handle it. Pointless code edits are not required.
